### PR TITLE
libarchive: update to 3.7.4

### PR DESCRIPTION
**Pull request by benoit-pierre**
The macOS build is fine on `x86_64`, but borked on `arm64`:
```
ld: library 'ssl' not found
```
Same issue when building curl (#1466).

### CODE REVIEW

**Comment by neheb** [comment]
brew bug?

**Comment by benoit-pierre** [comment]
> brew bug?

Not sure, looking into it.

> edit: does libarchive need libssl or just libcrypto?

Good catch, will amend to link with `libcrypto` only.

**Comment by benoit-pierre** [comment]
Looks like it is a brew bug:
```bash
# the include directory is correct:
▸ pkg-config --cflags libcrypto
-I/opt/homebrew/Cellar/openssl@3/3.3.0/include
# but the libraries search path is not:
▸ pkg-config --libs libcrypto
-L/opt/homebrew/Cellar/openssl@3/3.3.0 -lcrypto
#
▸ ls -l /opt/homebrew/Cellar/openssl@3/3.3.0
total 1752
-rw-r--r--   1 runner  admin     990 Apr  9 12:12 AUTHORS.md
-rw-r--r--   1 runner  admin  782003 Apr  9 12:12 CHANGES.md
-rw-r--r--   1 runner  admin    1256 Apr 22 18:13 INSTALL_RECEIPT.json
-rw-r--r--   1 runner  admin   10175 Apr  9 12:12 LICENSE.txt
-rw-r--r--   1 runner  admin   84478 Apr  9 12:12 NEWS.md
-rw-r--r--   1 runner  admin    8182 Apr  9 12:12 README.md
drwxr-xr-x   4 runner  admin     128 Apr 22 18:13 bin
drwxr-xr-x   3 runner  admin      96 Apr  9 12:12 include
drwxr-xr-x  12 runner  admin     384 Apr 22 18:13 lib
drwxr-xr-x   4 runner  admin     128 Apr  9 12:12 share
```

**Comment by benoit-pierre** [comment]
Reinstalling openssl fix it. ¯\\_(ツ)_/¯

File: .github/workflows/sanity_checks.yml [hunk 1]
```diff
@@ -154,6 +154,12 @@ jobs:
           brew install ninja
           python3 -m pip install --pre meson
 
+      # The pkg-config file for openssl is borked: the libraries search
```
Comment (neheb) [inline .github/workflows/sanity_checks.yml:157]: I think disable openssl in ci_config.json with this comment.

Comment (benoit-pierre) [reply .github/workflows/sanity_checks.yml:157]: Rather than fix it for everybody?

Comment (benoit-pierre) [reply .github/workflows/sanity_checks.yml:157]: If we wait a couple of days, maybe it won't be an issue anymore: https://github.com/actions/runner-images/issues/9755#issuecomment-2081579531.

Comment (neheb) [reply .github/workflows/sanity_checks.yml:157]: sure. I don't know if editing sanity_checks.yml for a specific Homebrew bug is the correct solution here.

Comment (benoit-pierre) [reply .github/workflows/sanity_checks.yml]: So it's still not fixed…

I disagree that disabling openssl in `ci_config.json` is the right approach. It's a dependency of at least 8 wraps, IMHO fixing it in `sanity_checks.yml` so all wraps benefit from it is the better solution.

Alternatively, ignore the macOS arm64 failure and merge.

Comment (neheb) [reply .github/workflows/sanity_checks.yml]: Yeah why not. It's experimental anyways.

**Pull request merged**